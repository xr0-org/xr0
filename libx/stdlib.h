#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
