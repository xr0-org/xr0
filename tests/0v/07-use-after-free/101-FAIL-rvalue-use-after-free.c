#include <stdlib.h>

void
func()
{
	int *p;
	int q;

	p = malloc(1);
	free(p);

	q = *p;		/* ERROR: undefined indirection (rvalue) */
}
