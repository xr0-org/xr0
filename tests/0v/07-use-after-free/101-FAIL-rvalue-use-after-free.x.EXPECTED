07-use-after-free/101-FAIL-rvalue-use-after-free.x:12:8: undefined memory access: *(p) has no value (func)
